src/r2mm/profiles/ProfilesClient.ts

```diff
@@ -0,0 +1,34 @@
+import Axios, { AxiosResponse } from 'axios';
+import R2Error from 'src/model/errors/R2Error';
+
+export default class ProfilesClient {
+    public static TSCreateAPI = 'https://thunderstore.io/api/experimental/legacyprofile/create/'
+    public static TSRetrieveAPI = (profileImportCode: string) => {return `https://thunderstore.io/api/experimental/legacyprofile/get/${profileImportCode}/`}
+    public static HerokuRetrieveAPI = (profileImportCode: string) => {return `https://r2modman-hastebin.herokuapp.com/raw/${profileImportCode}`}
+
+    public static async sendProfile(endpoint: string, payload: string): Promise<AxiosResponse | R2Error> {
+        try {
+            return await Axios.post(
+                endpoint,
+                payload,
+                {
+                    headers: {'Content-Type': 'application/octet-stream'}
+                }
+            )
+        } catch (e: any) {
+            if (Axios.isAxiosError(e)) {
+                if (e.response!.status == 429) {
+                    return new R2Error('Too many exports in a short period of time', ':PeepoShake:', 'Slow down there champ');
+                } else {
+                    return new R2Error('Failed to export profile', e.message, `${e.response!.status}`);
+                }
+            } else {
+                return new R2Error('Unexpected error', e.message, null);
+            }
+        }
+    }
+
+    public static async retrieveProfile(endpoint: string): Promise<AxiosResponse> {
+        return await Axios.get(endpoint)

```

Comment:
Semicolon missing, but also you can remove `async` from the function and `await` from this line and it'll work exactly as it does now (since Axios already returns a `Promise<AxiosResponse>` and you can just pass it forward). Semicolon also missing.